core/src/main/java/aima/core/agent/basic/Environment.java

```diff
@@ -0,0 +1,33 @@
+package aima.core.agent.basic;
+
+import aima.core.agent.api.Agent;
+import java.util.List;
+
+/**
+ * Possible environments in which Agent(s) can percieve and act.
+ * 
+ * @author Ritwik Sharma
+ */
+public interface Environment {
+	
+	/**
+	 * Returns the Agent(s) belonging to this Environment.
+	 */
+	List<Agent> getAgents();
```

Comment:
Agent in `AIMA4e` is a generic interface with a different API than in `AIMA3e`. Any implementation of this interface would possibly suffer from a number of unchecked warnings. Please verify the implementation by giving a supporting example environment that implements this interface.